compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -412,7 +412,7 @@ pub fn simplify_locals<'tcx>(body: &mut Body<'tcx>, tcx: TyCtxt<'tcx>) {
     if map.iter().any(Option::is_none) {
         // Update references to all vars and tmps now
         let mut updater = LocalUpdater { map, tcx };
-        updater.visit_body(body);
+        updater.visit_body_preserves_cfg(body);
```

Comment:
The control flow graphs is currently invalidated in `remove_unused_definitions` call a few lines earlier. That could be changed as well.